File: Lab-5/selectionSort.py

import time

def selectionSort(arr,n):
  for i in range(n):
    least = arr[i]
    pos = i
    for j in range(i+1,n):
      if arr[j] < least:
        least = arr[j]
        pos = j
    if i != pos:
      arr[i],arr[pos] = arr[pos],arr[i]
  return arr

arr = [22,67,32,45,12,107,678,4,66]
startTime = time.time()
n=len(arr)
sortedArray=selectionSort(arr,n)
endTime = time.time()
print ("Sorted array is:", sortedArray)
print ("Time taken to sort the array is: ", endTime-startTime)

File: Lab-5/selectionSort.c
#include <stdio.h>
#include <stdlib.h>
#include <time.h>
void SelectionSort(int A[], int n)
{
  int i, j, pos, least;
  for (i = 0; i < n; i++)
  {
    least = A[i];
    pos = i;
    for (j = i + 1; j < n; j++)
    {
      if (A[i] < least)
      {
        least = A[j];
        pos = j;
      }
    }
  }
}
void display(int A[], int n)
{
  int i;
  for (i = 0; i < n; i++)
  {
    printf("%d ", A[i]);
  }
  printf("\n");
}
int main()
{
  int A[100000], n, i;
  clock_t start, end, diff;
  srand(time(NULL));
  printf("Enter n ");
  scanf("%d", &n);
  for (i = 0; i < n; i++)
    A[i] = rand();
  display(A, n);
  start = clock();
  SelectionSort(A, n);
  end = clock();
  display(A, n);
  diff = end - start;
  printf("The time taken is %f second\n", (float)diff / CLOCKS_PER_SEC);
  return 0;
}